### PR TITLE
Progress bar

### PR DESCRIPTION
**Pull request by esauerbo**
Created a progress bar component that tracks and displays the current position in the song. Currently unstyled.

### CODE REVIEW

**Comment by doorlay** [comment]
Looks good! Approving...

**Review by doorlay — APPROVED** [review]

**Comment by Nathanllee1** [comment]
Ship it!

**Comment by BSilton26** [comment]
Looks good to me, but definitely could use some styling. Maybe match the color to the general styling of the whole web app? Like that same orange shade in that background pattern. I also noticed that changing the song progress in Spotify isn't reflected in Tabify, worth looking into.

File: spotify-web-player/src/ProgressBar.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      clearInterval(interval);
+      return;
+    }, 100);
+    return () => {
+      clearInterval(interval);
+    };
+  };
+  useEffect(timer, [position, is_paused]);
+
+  return <progress max={duration} value={position}></progress>;
```
Comment (BSilton26) [inline spotify-web-player/src/ProgressBar.js:21]: So is this just using a timer to track progress, not actually getting the real song progress?

**Review by BSilton26 — COMMENTED** [review]
Kinda confused on how progress works, but can clarify Wednesday 👍

File: spotify-web-player/src/ProgressBar.js (hunk 1 above)
Comment (esauerbo) [reply spotify-web-player/src/ProgressBar.js:21]: These aren't mutually exclusive, it is using a timer but it i still tracking the real song progress.

**Comment by esauerbo** [comment]
@BSilton26 thanks for the input. I realized that the since position is tracked in Webplackback.js, when it updates (ie skipping around within a song) the state of the progress bar wasn't being updated in time because the useEffect hook isn't immediate. Moving the progress bar into Webplayback will solve this, so I'll make a new PR.

**Comment by esauerbo** [comment]
@Nathanllee1 @doorlay 
Thanks!!

**Pull request merged**